src/core/lv_event.c

```diff
@@ -419,20 +431,41 @@ static lv_res_t event_send_core(lv_event_t * e)
     lv_indev_t * indev_act = lv_indev_get_act();
     if(indev_act) {
         if(indev_act->driver->feedback_cb) indev_act->driver->feedback_cb(indev_act->driver, e->code);
+        if(e->stop_processing) return LV_RES_OK;
         if(e->deleted) return LV_RES_INV;
     }
 
     lv_res_t res = LV_RES_OK;
-    res = lv_obj_event_base(NULL, e);
-
     lv_event_dsc_t * event_dsc = res == LV_RES_INV ? NULL : lv_obj_get_event_dsc(e->current_target, 0);
 
     uint32_t i = 0;
     while(event_dsc && res == LV_RES_OK) {
-        if(event_dsc->cb && (event_dsc->filter == LV_EVENT_ALL || event_dsc->filter == e->code)) {
+        if(event_dsc->cb  && ((event_dsc->filter & LV_EVENT_PREPROCESS) == LV_EVENT_PREPROCESS)
+           && (event_dsc->filter == LV_EVENT_ALL || (event_dsc->filter & ~LV_EVENT_PREPROCESS) == e->code)) {
```

Comment:
Doesn't it need to be `(event_dsc->filter & ~LV_EVENT_PREPROCESS) == LV_EVENT_ALL`?

Reply:
Oops. You're right.